src/geneweaver/api/controller/genesets.py

```diff
@@ -29,19 +37,73 @@ def get_geneset(
     geneset_id: int,
     user: UserInternal = Security(deps.full_user),
     cursor: Optional[deps.Cursor] = Depends(deps.cursor),
+    # using int type and adding doc options here as using GeneIdentifier
+    # won't produce the right docs for gene identifier options
+    gene_id_type: Optional[int] = Query(
+        None, description=f"Options: {gene_id_type_options}"
+    ),
 ) -> dict:
-    """Get a geneset by ID."""
-    response = genset_service.get_geneset(cursor, geneset_id, user)
+    """Get a geneset by ID. Optional filter results by gene identifier type."""
+    if gene_id_type:
+        gene_identifier_type = get_gene_identifier_type(gene_id_type)
+        response = genset_service.get_geneset_w_gene_id_type(
+            cursor, geneset_id, user, gene_identifier_type
+        )
+    else:
+        response = genset_service.get_geneset(cursor, geneset_id, user)
 
     if "error" in response:
-        if response.get("message") == api_message.ACCESS_FORBIDEN:
-            raise HTTPException(status_code=403, detail=api_message.ACCESS_FORBIDEN)
+        if response.get("message") == api_message.ACCESS_FORBIDDEN:
+            raise HTTPException(status_code=403, detail=api_message.ACCESS_FORBIDDEN)
         else:
             raise HTTPException(status_code=500, detail=api_message.UNEXPECTED_ERROR)
 
     return response
 
 
+@router.get("/{geneset_id}/export", response_class=FileResponse)
```

Comment:
I don't know if I love the word `export` here. I think I'd prefer `geneset/{geneset_id}/file`. Curious what @gerring thinks.